src/main/java/com/minecolonies/client/gui/WindowBuildTool.java

```diff
@@ -367,25 +367,34 @@ private void changeSchematic()
         String labelHutDec = findPaneOfTypeByID(BUTTON_HUT_DEC_ID, Button.class).getLabel();
         String labelHutStyle = findPaneOfTypeByID(BUTTON_STYLE_ID, Button.class).getLabel();
 
-        @NotNull SchematicWrapper schematic = new SchematicWrapper(this.mc.theWorld, labelHutStyle + '/' + labelHutDec + (Settings.instance.isInHutMode() ? (level + 1) : ""));
+        String resourceName = labelHutStyle + '/' + labelHutDec + (Settings.instance.isInHutMode() ? (level + 1) : "");
+        try
+        {
 
-        Settings.instance.setActiveSchematic(schematic.getSchematic());
+            @NotNull SchematicWrapper schematic = new SchematicWrapper(this.mc.theWorld, resourceName);
 
-        Settings.instance.moveTo(this.pos);
+            Settings.instance.setActiveSchematic(schematic.getSchematic());
 
-        //Catch up on rotations, makes it so going up a level or changing style doesn't reset rotation.
-        if (this.rotation == ROTATE_LEFT)
-        {
-            RotationHelper.rotate(Settings.instance.getSchematicWorld(), EnumFacing.DOWN, true);
-        }
-        else
-        {
-            //Runs 0, 1, or 2 times.
-            for (int times = 0; times < rotation; times++)
+            Settings.instance.moveTo(this.pos);
+
+            //Catch up on rotations, makes it so going up a level or changing style doesn't reset rotation.
+            if (this.rotation == ROTATE_LEFT)
             {
-                RotationHelper.rotate(Settings.instance.getSchematicWorld(), EnumFacing.UP, true);
+                RotationHelper.rotate(Settings.instance.getSchematicWorld(), EnumFacing.DOWN, true);
+            }
+            else
+            {
+                //Runs 0, 1, or 2 times.
+                for (int times = 0; times < rotation; times++)
+                {
+                    RotationHelper.rotate(Settings.instance.getSchematicWorld(), EnumFacing.UP, true);
+                }
             }
         }
+        catch(java.lang.IllegalStateException err)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1166)